lang/test/org/partiql/lang/thread/ThreadInterruptedTests.kt

```diff
@@ -3,6 +3,12 @@ package org.partiql.lang.thread
 
```

Comment:
Think we also need to:
1. Update CHANGELOG.md w/ these changes
2. Update PR description with more concrete benchmarking results
3. Include why interruption and exception handling was added -- I know there was some offline discussion but would be good to capture it in the PR

Reply:
Done. LMK if that works.

Reply:
Thanks, changes look good